exec/kubernetes_exec_manager.go

```diff
@@ -67,6 +67,84 @@ func NewK8sExecManager(
 	}
 }
 
+func (manager *KubernetesExecManager) Resolve(container, token string) (*model.ResolvedExec, error) {
+	machineExec := &model.MachineExec{
+		BearerToken: token,
+	}
+	k8sAPI, err := manager.k8sAPIProvider.GetK8sAPI(machineExec)
+	if err != nil {
+		logrus.Debugf("Unable to get k8sAPI %s", err.Error())
+		return nil, err
+	}
+	logrus.Debug("Successfully Got k8sApi object.")
+
+	if container != "" {
+		containerFilter := filter.NewKubernetesContainerFilter(manager.namespace, k8sAPI.GetClient().CoreV1())
+		containerInfo, err := containerFilter.GetContainer(container)
+		if err != nil {
+			return nil, err
+		}
+		cmdResolver := NewCmdResolver(k8sAPI, manager.namespace)
+		resolvedCmd, err := cmdResolver.ResolveCmd(*machineExec, containerInfo)
+
+		if err != nil {
+			return nil, err
+		}
+
+		logrus.Printf("%s is successfully resolved in auto discovered container %s/%s", resolvedCmd,
+			containerInfo.PodName, containerInfo.ContainerName)
+		return &model.ResolvedExec{
+			PodName:       containerInfo.PodName,
+			ContainerName: containerInfo.ContainerName,
+			Cmd:           resolvedCmd,
+		}, nil
+	} else {
```

Comment:
This `else` introduces quite a bit of copied code -- I think we can refactor it out but maybe it's better to leave for a separate PR.